lib/src/Uncertainty/Distribution/ConditionedDistribution.cxx

```diff
@@ -0,0 +1,591 @@
+    const Scalar cxx = cov(nonConditioningIndices_[0], nonConditioningIndices_[0]);
+    const Scalar cxy = cov(1, 0);
+    const Scalar muy = mu[conditioningIndices_[0]];
+    const Scalar mux = mu[nonConditioningIndices_[0]];
+    const Scalar y = conditioningValues_[0];
+    Point muConditioned(getDimension());
+    CovarianceMatrix covConditioned(getDimension());
+    muConditioned[0] = mux + cxy / cyy * (y - muy);
+    covConditioned(0, 0) = cxx - cxy / cyy * cxy;
+    simplified = Normal(muConditioned, covConditioned);
+    return true;
+  }
+  return false;
+}
+
+Distribution ConditionedDistribution::getSimplifiedVersion() const
+{
+  return useSimplifiedVersion_ ? simplifiedVersion_ : *this;
+}
+
+/* Compute the numerical range of the distribution given the parameters values */
+void ConditionedDistribution::computeRange()
+{
+  if (useSimplifiedVersion_)
+    setRange(simplifiedVersion_.getRange());
+
+  setRange(marginalConditionedDistribution_.getRange());
+}
+
+/* Get the support of a distribution that intersect a given interval */
+Sample ConditionedDistribution::getSupport(const Interval & interval) const
+{
+  if (interval.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given interval has a dimension that does not match the distribution dimension.";
+  if (!isDiscrete()) throw NotDefinedException(HERE) << "Error: the support is defined only for discrete distributions.";
+  Collection<Sample> marginalSupport(getDimension());
+  Indices marginalSize(getDimension());
+  for (UnsignedInteger j = 0; j < getDimension(); ++ j)
+  {
+    marginalSupport[j] = marginalConditionedDistribution_.getMarginal(j).getSupport(interval.getMarginal(j));
+    marginalSize[j] = marginalSupport[j].getSize();
+  }
+  IndicesCollection tuples(Tuples(marginalSize).generate());
+  Sample support(0, getDimension());
+  for (UnsignedInteger i = 0; i < tuples.getSize(); ++ i)
+  {
+    Point x(getDimension());
+    for (UnsignedInteger j = 0; j < getDimension(); ++ j)
+      x[j] = marginalSupport[j](tuples(i, j), 0);
+    support.add(x);
+  }
+  return support;
+}
+
+/* Get one realization of the distribution */
+Point ConditionedDistribution::getRealization() const
+{
+  if (useSimplifiedVersion_)
+    return simplifiedVersion_.getRealization();
+
+  if (distribution_.hasIndependentCopula())
+    return distribution_.getRealization().select(nonConditioningIndices_);
+  else
+    return DistributionImplementation::getRealization();
+}
+
+
+Point ConditionedDistribution::expandPoint(const Point & point) const
+{
+  const UnsignedInteger dimension = getDimension();
+  if (point.getDimension() != dimension) throw InvalidArgumentException(HERE) << "Error: expected a point of dimension=" << dimension << ", got dimension=" << point.getDimension();
+  Point fullPoint(distribution_.getDimension());
+  UnsignedInteger indexC = 0;
+  UnsignedInteger indexP = 0;
+  for (UnsignedInteger i = 0; i < distribution_.getDimension(); ++ i)
```

Comment:
I used some Numpy masking at:

https://github.com/openturns/otbenchmark/blob/ae19ef309f7d75ee249687f0802331f8b84f3c87/otbenchmark/_ConditionalDistribution.py#L337

Isn't there some generic C++ masking feature to implement this? This could be implemented through a new `ConditionalPoint` class, which could then used elsewhere, e.g. in `ParametricFunction`, where the same code seems to be implemented:

https://github.com/openturns/openturns/blob/88adceb3afc5a0b515323ec06d46cf69503ada6c/lib/src/Base/Func/ParametricEvaluation.cxx#L102

Reply:
I dont thinks that exists

Reply:
See ValArray and std::iota in C++ 11.

Reply:
the new version is much better now